docs/source/contributors/extending_fitbenchmarking.rst

```diff
@@ -43,21 +42,58 @@ respectively.
 
 **Adding new fitting problem definition types**
 
-Follow the following steps
-
-1. Create ``parse_{type}_data.py`` which
-   contains a child class of ``BaseFittingProblem`` in
-   ``parsing/base_fitting_problem.py`` that processes the type (format) and
-   initialise the class with appropriate attributes (examples can be found
-   in ``parse_{nist/fitbenchmark/sasview}_data.py``).
-   As a minimum this must implement the abstract get_function method which
-   returns a list of callable-initial parameter pairs.
-2. In ``parsing/parse.py``
-   alter the function ``determine_problem_type()`` such that it determines
-   the new type
-3. In ``parsing/parse.py`` add a new if statement to
-   ``parse_problem_file()`` to call the user defined
-   ``parse_{type}_data.py`` class
+To add a new fitting problem type, it is a requirement that the parser name
+can be derived from the file to be parsed.
+This is done for all current file formats by including it as the first line
+in the file. e.g ``# Fitbenchmark Format`` or ``NIST/ITL StRD``.
```

Comment:
I'm not sure if the NIST format is autogenerated somewhere... whereas it looks like the SAS and fitbenchmarking ones were both a homebrew format

Reply:
The NIST data format is just what they use to distribute the NIST nonlinear regression test set -- see, e.g., the Data File link at the top of here:
https://www.itl.nist.gov/div898/strd/nls/data/misra1c.shtml

Realistically, no new tests are going to be added to this (this test set has been static for decades), so changing the header wouldn't be a problem if that would be cleaner.

Reply:
Discussed today at the meeting and I think the consensus was to leave NIST problems as they are since we didn't create the files. (Could possibly change CuteST, as that was hand written but that it wouldn't match).

Reply:
Fair enough, I have no problems with that.